packages/backend/src/clients/service.ts

```diff
@@ -0,0 +1,78 @@
+import { TransactionOrKnex } from 'objection'
+
+import { Client } from './model'
+import { BaseService } from '../shared/baseService'
+import { ClientKeys } from '../clientKeys/model'
+import { JWKWithRequired } from 'auth'
+
+export interface CreateClientOptions {
+  name: string
+  uri: string
+}
+
+export interface AddKeyToClientOptions {
+  id: string
+  clientId: string
+  jwk: JWKWithRequired
+}
+
+export interface ClientService {
+  getClient(id: string): Promise<Client | undefined>
+  createClient(options: CreateClientOptions): Promise<Client>
+  addKeyToClient(options: AddKeyToClientOptions): Promise<Client>
+}
+
+interface ServiceDependencies extends BaseService {
+  knex: TransactionOrKnex
+}
+
+export async function createClientService({
+  logger,
+  knex
+}: ServiceDependencies): Promise<ClientService> {
+  const log = logger.child({
+    service: 'ClientKeysService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex
+  }
+  return {
+    getClient: (id) => getClient(deps, id),
+    createClient: (options) => createClient(deps, options),
+    addKeyToClient: (options) => addKeyToClient(deps, options)
+  }
+}
+
+async function getClient(
+  deps: ServiceDependencies,
+  id: string
+): Promise<Client | undefined> {
+  const client = await Client.query(deps.knex).findById(id)
+  client.keys = await getClientKeys(deps, client.id)
```

Comment:
Does something like this work?
https://vincit.github.io/objection.js/api/query-builder/eager-methods.html#withgraphfetched
```suggestion
  const client = await Client.query(deps.knex).findById(id).withGraphFetched('keys')
```

Reply:
It does. But I was trying to `orderBy` which was getting quite complicated to do with the relation mappings. And this is only used by the admin api so I thought it probably wasn't necessary.